compiler/ballerina-lang/src/main/java/io/ballerina/projects/DependencyGraph.java

```diff
@@ -60,6 +60,30 @@ private DependencyGraph(Map<T, Set<T>> dependencies) {
     public void findCycles() {
     }
 
+    /**
+     * Compares two instances of {@code DependencyGraph}.
+     *
+     * @param other other dependency graph to compare with
+     * @return DependencyGraph.Compatibility.COMPATIBLE if graphs are compatible;
+     *         DependencyGraph.Compatibility.INCOMPATIBLE otherwise
+     */
+    public Compatibility compareTo(DependencyGraph<T> other) {
```

Comment:
Let `O` be the set of nodes in the old dependency graph and `N` be the new dependency graph set. So you are checking whether `O-N` is empty to figure out whether there are any deleted nodes. The logic is correct; however, I don't think this method name and return type make sense.

1) Shall we not make this a public method for now? This is only used for internal implementation at the moment
2) We can rename this method as `difference` and change the return signature to return a collection of nodes.

Reply:
Addressed in 7b14ab4